### PR TITLE
INTSAMPLES-144: Barrier Sample

### PR DESCRIPTION
**Pull request by garyrussell**
JIRA: https://jira.spring.io/browse/INTSAMPLES-144

### CODE REVIEW

**Comment by garyrussell** [comment]
Pushed (switched to boot).

File: basic/barrier/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+Barrier Sample
+==============
+
+This example demonstrates the use of a process barrier component to suspend a thread until some asynchronous operation 
+completes. It uses an **HTTP Inbound Gateway**, splits the request, sends the splits to rabbitmq and then waits for
```
Comment (artembilan) [inline basic/barrier/README.md]: Minor: new sentence from new line.

Comment (artembilan) [inline basic/barrier/README.md]: Or never mind: this is `markdown`, not `asciidoc` :smile:

File: build.gradle [hunk 2]
```diff
@@ -437,6 +437,25 @@ project('amqp') {
...
+	description = 'Barrier Sample'
+
+	apply plugin: 'spring-boot'
+
+	dependencies {
+		compile 'org.springframework.boot:spring-boot-starter-web'
+		compile 'org.springframework.boot:spring-boot-starter-integration'
+		compile "org.springframework.integration:spring-integration-core:4.2.0.BUILD-SNAPSHOT"
+		compile "org.springframework.integration:spring-integration-amqp:4.2.0.BUILD-SNAPSHOT"
+		compile "org.springframework.integration:spring-integration-http:4.2.0.BUILD-SNAPSHOT"
```
Comment (artembilan) [inline build.gradle]: `$springIntegrationVersion` is more feasible here.
Although I don't think that we should be in a hurry and there is just enough to wait for the proper Boot version to pick up those SI changes.
WDYT?

Comment (garyrussell) [inline build.gradle]: Yes; we can wait

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration-samples/commit/db3d78109fe614c3e398fd357be8d6ef473f8473

**Pull request closed without merging**